### PR TITLE
Enable Android x86_64 support

### PR DESCRIPTION
**Pull request by crtrott**
From Chris: 

Special thanks to @crtrott for the thrilling late-night pair programming session.

As a note, since Kokkos requires 64 bit pointers, it will not build with Android x86 support, so you'd need to turn that off in your Android build.

Tested in Pixel 5 emulator

### CODE REVIEW

File: cmake/kokkos_compiler_id.cmake [hunk 1]
```diff
@@ -218,4 +218,13 @@ ENDIF()
...
+LIST(LENGTH VERSION_LIST LIST_LENGTH)
+
+# On Android, the compiler doesn't have a patch version, just a major/minor
+IF(LIST_LENGTH GREATER 2)
+  LIST(GET VERSION_LIST 2 KOKKOS_COMPILER_VERSION_PATCH)
+ELSE()
+  SET(KOKKOS_COMPILER_VERSION_PATCH 0)
+ENDIF()
+
+
```
Comment (dalg24) [inline cmake/kokkos_compiler_id.cmake:230]:
```suggestion
```
(Trailing whitespace on unnecessary line break)

File: core/src/Kokkos_Macros.hpp [hunk 2]
```diff
@@ -612,7 +612,7 @@ static constexpr bool kokkos_omp_on_host() { return false; }
 
 #if (defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG) ||  \
      defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_PGI)) && \
-    !defined(_WIN32)
+  !defined(_WIN32) && !defined(__ANDROID__)
```
Comment (dalg24) [inline core/src/Kokkos_Macros.hpp:615]:
```suggestion
    !defined(_WIN32) && !defined(__ANDROID__)
```

**Review by dalg24 — COMMENTED** [review]
Looks fine besides the formatting

**Comment by csiefer2** [comment]
Maybe I'll figure out how to use the clang formatter in Android Studio next time :)

**Comment by dalg24** [comment]
Failures are clearly unrelated (machine out of disk space)

**Review by dalg24 — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Review by JBludau — APPROVED** [review]

**Pull request merged**